flake.nix

```diff
@@ -0,0 +1,55 @@
+{
+  description = "Simon’s Improved Layout Engine";
+
+  # To make user overrides of the nixpkgs flake not take effect
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+
+  # https://nixos.wiki/wiki/Flakes#Using_flakes_project_from_a_legacy_Nix
+  inputs.flake-compat = {
+    url = "github:edolstra/flake-compat";
+    flake = false;
+  };
+
+  outputs = { self
+    , nixpkgs
+    , flake-utils
+    , flake-compat
+  }:
+  flake-utils.lib.eachDefaultSystem (system:
+    let
+      pkgs = import nixpkgs {
+        inherit system;
+      };
+      # Use the expression from Nixpkgs instead of rewriting it here.
+      sile = pkgs.sile.overrideAttrs(oldAttr: {
+        version = "${(pkgs.lib.importJSON ./package.json).version}-flake";
+        # TODO: Building from source doesn't work yet as the fonts are not
+        # available during build and can't be downloaded in the sandbox.
+        src = builtins.filterSource
+          (path: type: type != "directory" || baseNameOf path != ".git")
+          ./.;
+        nativeBuildInputs = oldAttr.nativeBuildInputs ++ [
+          pkgs.autoreconfHook
+        ];
+        # TODO: This change can be updated in nixpkgs on the next version bump
+        postPatch = oldAttr.preConfigure;
```

Comment:
thought: could it make sense to try and modify .gitmodules file to have a specific revision of libtexpdf instead of master, then try to extract the revision from `builtins.readFile ./.gitmodules` and fetchGitHub it and copy into ./libtexpdf directory?